Assets/Mirror/Editor/Weaver/Weaver.cs

```diff
@@ -385,25 +387,33 @@ static bool CheckMessageBase(TypeDefinition td)
 
             bool didWork = false;
 
-            // are ANY parent classes MessageBase
-            TypeReference parent = td.BaseType;
-            while (parent != null)
+            if (td.ImplementsInterface(IMessageBaseType))
             {
-                if (parent.FullName == MessageBaseType.FullName)
-                {
-                    MessageClassProcessor.Process(td);
-                    didWork = true;
-                    break;
-                }
-                try
-                {
-                    parent = parent.Resolve().BaseType;
-                }
-                catch (AssemblyResolutionException)
+                MessageClassProcessor.Process(td);
+                didWork = true;
+            }
+            else
+            {
+                // are ANY parent classes MessageBase
+                TypeReference parent = td.BaseType;
+                while (parent != null)
                 {
-                    // this can happen for plugins.
-                    //Console.WriteLine("AssemblyResolutionException: "+ ex.ToString());
-                    break;
+                    if (parent.FullName == MessageBaseType.FullName)
```

Comment:
Why not check if parent implements the interface instead?

Reply:
i tried to keep existing code as much as possible, so i didn't touch that